sb3_contrib/common/maskable/utils.py

```diff
@@ -0,0 +1,31 @@
+from typing import Callable, Optional
+
+import gym
+import numpy as np
+from stable_baselines3.common.type_aliases import GymEnv
+from stable_baselines3.common.vec_env import VecEnv
+
+
+def get_action_masks(env: GymEnv) -> np.ndarray:
+    method_name = "action_masks"
+
+    if isinstance(env, VecEnv):
+        return np.stack(env.env_method(method_name))
+    else:
+        return getattr(env, method_name)()
+
+
+def get_action_mask_fn(env: GymEnv) -> Optional[Callable[[gym.Env], np.ndarray]]:
+    attr_name = "action_mask_fn"
+
+    try:
+        if isinstance(env, VecEnv):
+            return env.get_attr(attr_name)[0]
+        else:
+            return getattr(env, attr_name)
+    except AttributeError:
+        return None
+
+
+def is_masking_supported(env: GymEnv) -> bool:
+    return get_action_mask_fn(env) is not None
```

Comment:
I'd thought looking for the `action_masks` method would be enough (no need to call it, just make sure it's there). Do you really need to check for the `fn`  there, and latter on just use the other method ? There must be a way to go with just one.

At the moment if I only define `action_masks`, it (seems to) work properly, but spit a (wrong?) warning about my Env not supporting masking ...

Reply:
Okay, so it seems you need the `fn` to wrap other Envs (like the evaluation one in `MaskableEvalCallback`).

In My env I just defined

``` py
class MyEnv(gym.Env):
    ...
    def action_masks(self):
        ...
    action_mask_fn = action_masks
```

And that did it. So one of them really seems to be too much.

Reply:
Thanks for the feedback. I've simplified the API, there's no longer an expectation that envs will have an `action_mask_fn` attribute.